GCC warnings building the 5.0.1.rc3 tarball

These may be false positives, but just in case, I'm reporting the warnings...

```console
$ gcc --version
gcc (GCC) 15.2.1 20260123 (Red Hat 15.2.1-7)
```

---

```
/project/package/source/src/mpi/coll/barrier/barrier_intra_k_dissemination.c: In function ‘MPIR_Barrier_intra_k_dissemination’:
/project/package/source/src/mpi/coll/barrier/barrier_intra_k_dissemination.c:120:9: warning: ‘MPIC_Waitall’ accessing 8 bytes in a region of size 0 [-Wstringop-overflow=]
  120 |         MPIC_Waitall(k - 1, recv_reqs + ((k - 1) * ((nphases - 1) & 1)), MPI_STATUSES_IGNORE);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/project/package/source/src/mpi/coll/barrier/barrier_intra_k_dissemination.c:120:9: note: referencing argument 2 of type ‘MPIR_Request *[0]’
In file included from /project/package/source/src/include/mpiimpl.h:208,
                 from /project/package/source/src/mpi/coll/barrier/barrier_intra_k_dissemination.c:6:
/project/package/source/src/include/mpir_coll.h:48:5: note: in a call to function ‘MPIC_Waitall’
   48 | int MPIC_Waitall(int numreq, MPIR_Request * requests[], MPI_Status * statuses);
      |     ^~~~~~~~~~~~
```

---

```
In file included from /project/package/source/src/mpl/include/mpl_atomic_c11.h:20,
                 from /project/package/source/src/mpl/include/mpl_atomic.h:101,
                 from /project/package/source/src/mpl/include/mpl.h:16,
                 from /project/package/source/src/include/mpiimpl.h:53,
                 from /project/package/source/src/mpi/attr/attrutil.c:6:
In function ‘MPL_atomic_release_store_int’,
    inlined from ‘MPID_Attr_alloc’ at /project/package/source/src/mpi/attr/attrutil.c:48:5,
    inlined from ‘MPIR_Attr_dup_list’ at /project/package/source/src/mpi/attr/attrutil.c:167:17:
/project/package/source/src/mpl/include/mpl_atomic_c11.h:59:5: warning: ‘__atomic_store_4’ writing 4 bytes into a region of size 0 overflows the destination [-Wstringop-overflow=]
   59 |     atomic_store_explicit(&ptr->v, (CAST_TYPE)val, memory_order_release);      \
      |     ^~~~~~~~~~~~~~~~~~~~~
/project/package/source/src/mpl/include/mpl_atomic_c11.h:93:9: note: in expansion of macro ‘MPLI_ATOMIC_DECL_FUNC_COMMON’
   93 |         MPLI_ATOMIC_DECL_FUNC_COMMON(TYPE, NAME, ATOMIC_TYPE, CAST_TYPE) \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/project/package/source/src/mpl/include/mpl_atomic_c11.h:99:1: note: in expansion of macro ‘MPLI_ATOMIC_DECL_FUNC_VAL’
   99 | MPLI_ATOMIC_DECL_FUNC_VAL(int, int, atomic_int, int)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘MPIR_Attr_dup_list’:
cc1: note: destination object is likely at address zero
```

---

```
/project/package/source/src/mpid/ch4/src/ch4_proc.c: In function ‘MPIDIU_upidhash_add’:
/project/package/source/src/mpid/ch4/src/ch4_proc.c:336:5: warning: ‘memcpy’ specified bound between 18446744071562067968 and 18446744073709551615 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]
  336 |     memcpy(t->upid, upid, upid_len);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/include/mpir_coll.h
+++ b/src/include/mpir_coll.h
@@ -57,7 +57,7 @@ int MPIC_Isend(const void *buf, MPI_Aint count, MPI_Datatype datatype, int dest,
                MPIR_Comm * comm_ptr, MPIR_Request ** request, int coll_attr);
 int MPIC_Irecv(void *buf, MPI_Aint count, MPI_Datatype datatype, int source,
                int tag, MPIR_Comm * comm_ptr, MPIR_Request ** request);
-int MPIC_Waitall(int numreq, MPIR_Request * requests[], MPI_Status * statuses);
+int MPIC_Waitall(int numreq, MPIR_Request ** requests, MPI_Status * statuses);
 
 int MPIR_Reduce_local(const void *inbuf, void *inoutbuf, MPI_Aint count, MPI_Datatype datatype,
                       MPI_Op op);

--- a/src/mpi/attr/attrutil.c
+++ b/src/mpi/attr/attrutil.c
@@ -44,8 +44,9 @@ MPIR_Attribute *MPID_Attr_alloc(void)
     MPIR_Attribute *attr = (MPIR_Attribute *) MPIR_Handle_obj_alloc(&MPID_Attr_mem);
     /* attributes don't have refcount semantics, but let's keep valgrind and
      * the debug logging pacified */
-    MPIR_Assert(attr != NULL);
-    MPIR_Object_set_ref(attr, 0);
+    if (attr) {
+        MPIR_Object_set_ref(attr, 0);
+    }
     return attr;
 }
 

--- a/src/mpid/ch4/src/ch4_proc.c
+++ b/src/mpid/ch4/src/ch4_proc.c
@@ -333,7 +333,8 @@ void MPIDIU_upidhash_add(const void *upid, int upid_len, MPIR_Lpid lpid)
     t = MPL_malloc(sizeof(MPIDI_upid_hash), MPL_MEM_OTHER);
     t->lpid = lpid;
     t->upid = MPL_malloc(upid_len, MPL_MEM_OTHER);
-    memcpy(t->upid, upid, upid_len);
+    /* the unsigned cast is to suppress the gcc warning when it believes upid_len can take negative values */
+    memcpy(t->upid, upid, (unsigned) upid_len);
     t->upid_len = upid_len;
     HASH_ADD_KEYPTR(hh, upid_hash, t->upid, upid_len, t, MPL_MEM_OTHER);
 

